src/main/java/ch/njol/skript/util/Utils.java

```diff
@@ -840,4 +856,53 @@ public int hashCode() {
 
 	}
 
+	/**
+	 * Prints a warning about the loading/use of a class that has been deprecated or removed.
+	 * This is a fairly-unsafe method and should only be used during class-loading.
+	 *
+	 * @param source The class about which to print the warning. This MUST be the class calling this method.
+	 * @return 0 (for use by interfaces)
+	 */
+	@ApiStatus.Internal
+	public static int loadedRemovedClassWarning(Class<?> source) {
+		Logger logger = Skript.getInstance().getLogger();
+		Exception exception = new Exception();
+		exception.fillInStackTrace();
+		StackTraceElement[] stackTrace = exception.getStackTrace();
+		StackTraceElement caller = stackTrace[2];
+		String authors, name;
+		try {
+			Class<?> callingClass = Class.forName(caller.getClassName());
+			JavaPlugin plugin = JavaPlugin.getProvidingPlugin(callingClass);
+			//noinspection UnstableApiUsage
+			name = plugin.getPluginMeta().getDisplayName();
+			//noinspection UnstableApiUsage
+			authors = String.valueOf(plugin.getPluginMeta().getAuthors());
```

Comment:
`getPluginMeta` is Paper API. We need to use the [PluginDescriptionFile](https://hub.spigotmc.org/javadocs/spigot/org/bukkit/plugin/PluginDescriptionFile.html).

```suggestion
			name = plugin.getDescription().getFullName();
			authors = String.valueOf(plugin.getDescription().getAuthors());
```